### PR TITLE
ib/ add validation and errors handling

### PR DESCRIPTION
**Pull request by ibonksoftserve**

### CODE REVIEW

File: backend/app/core/api.interface.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  NOT_FOUND = 404,
+  SERVER = 500,
+}
+
+// <------- Errors ------->
+export enum ErrorType {
+  BAD_TOKEN = 'BadTokenError',
+  TOKEN_EXPIRED = 'TokenExpiredError',
+  UNAUTHORIZED = 'AuthFailureError',
+  SERVER = 'ServerlError',
```
Comment (kapoliub) [inline backend/app/core/api.interface.ts]: check the string

Comment (kapoliub) [reply backend/app/core/api.interface.ts]: you can install the following extension 
streetsidesoftware.code-spell-checker

File: backend/app/modules/user/user.controller.ts [hunk 2]
```diff
@@ -15,90 +17,48 @@ export class UserController {
...
-      res.status(500).send(error);
-    }
+    return new SuccessResponse<IUserResponse>(user).send(res)
   }
 
-  public async updateUser(req: Request, res: Response): Promise<void> {
-    try {
-      const body = req.body;
-      const { id } = req.params;
+  public async createUser(req: Request, res: Response): Promise<Response> {
```
Comment (kapoliub) [inline backend/app/modules/user/user.controller.ts:36]: we need to check if new user with the same name exists

**Review by kapoliub — COMMENTED** [review]
GTM. just fix comments

File: backend/app/core/api.interface.ts (hunk 1 above)
Comment (ibonksoftserve) [reply backend/app/core/api.interface.ts]: good extension
thanks

File: backend/app/modules/user/user.controller.ts (hunk 2 above)
Comment (ibonksoftserve) [reply backend/app/modules/user/user.controller.ts:36]: will be added during full user model implementation

**Review by kapoliub — APPROVED** [review]

**Pull request merged**